Assets/Scripts3D/Weapons/Weapon.cs

```diff
@@ -43,7 +72,39 @@ private void Shoot()
         {
             return;
         }
+    }
+
+    void RotationSway()
+    {
+        float mouseX = Input.GetAxisRaw("Mouse X") * _swayMultiplier;
+        float mouseY = Input.GetAxisRaw("Mouse Y") * _swayMultiplier;
+
+        Quaternion rotationX = Quaternion.AngleAxis(-mouseY, Vector3.right);
+        Quaternion rotationY = Quaternion.AngleAxis(mouseX, Vector3.up);
 
+        Quaternion targetRotation = rotationX * rotationY;
+
+        transform.localRotation = Quaternion.Slerp(transform.localRotation, targetRotation, _smooth * Time.deltaTime);
+    }
 
+    void HeadBobController()
+    {
+        if(PlayerController.mouseX == 0 && PlayerController.mouseY == 0)
```

Comment:
You can use InputHelper for these

```suggestion
        if(InputHelper.VerticalAxis == 0 && InputHelper.HorizontalAxis == 0)
```

And I'd say it's not super clear why it's being done, so we could move this into a variable:

`var playerIsStationary = InputHelper.VerticalAxis == 0 && InputHelper.HorizontalAxis == 0;

if (playerIsStationary) ....`